src/Tables/molecules/TypeTable.tsx

```diff
@@ -49,29 +48,55 @@ interface IProps {
   hasStatus?: boolean
   hasThumbnail?: boolean
   hasTypeIcon?: boolean
+  initialSortColumnId?: boolean
   selectCallback? : (checked:boolean, id?: string | number)=>void
   toggleAllCallback? : (checked: boolean)=>void
+  sortCallback? : (ascending: boolean, columnId?: string) => void
 }
 
 const TypeTable : React.FC<IProps> = ({
   columnConfig,
   selectable,
-  selectCallback = ()=>{},
-  toggleAllCallback = ()=>{},
   rows = [],
   hasStatus = false,
   hasThumbnail = false,
-  hasTypeIcon = false
+  hasTypeIcon = false,
+  selectCallback = ()=>{},
+  toggleAllCallback = ()=>{},
+  sortCallback = ()=>{},
 }) => {
   const [allChecked, setAllChecked] = useState(false);
   const toggleAllCallbackWrapper = useCallback((checked:boolean) => {
     toggleAllCallback(checked);
   }, [toggleAllCallback]);
+  const [sortSpec, setSortSpec] = useState(columnConfig);
 
   useEffect(() => {
     setAllChecked(rows.every(isChecked) && rows.length > 0);
   }, [rows]);
 
+  /**
+   * toggleSort
+   * Toggle the clicked header
+   * Make clicked header active true and the rest false;
+   */
+  const toggleSort = useCallback((columnKey : number) => {
+    if(sortSpec[columnKey] === undefined) { return;}
+    const updatedSort = [...sortSpec]
+    updatedSort.forEach((col, key) => {
+      if(key === columnKey) {
+        col.sortActive = true;
+      } else {
+        col.sortActive = false;
+      }
+    });
+    const newAscending = undefined ? true: !sortSpec[columnKey].ascending;
+    updatedSort[columnKey].ascending = newAscending
+    const colId = (updatedSort[columnKey].columnId === undefined) ?  '' : updatedSort[columnKey].columnId;
```

Comment:
IMO this is a bit strange. If the the `colId` should either be the index or the user provided ID, how can the user know which column has changes otherwise.

Reply:
#### I agree that it looks strange.

The only reason I implemented like this is to allow columnId to be optional. ( we talk that columnId shouldn't be required if sorting is not required in the table being the main reason to prevent breaking previous implementations.)

**The only scenario** where I think the UI library user would use sorting and would be too lazy to provide a columnId  is if the table has only **one sorting column**. In that case the user would know what is sorting criteria.

If you want this part to be refactor I can think of two other ways to implement it.

#### **Option A** 
Each Column has it's own sortingCallback instead one sorting handler and the user would know what to sort because each implementation it's independent.

```js
const sortByName = (ascending: boolean) => { awesomeData being sorted by name}
const sortByUploadTime = (ascending: boolean) => {  awesomeData being sorted by uploadTime}

const columnConfigSample : ITableColumnConfig[] = [
  {
    columnId: 'deviceName',
    header: 'Device Name',
    sortable: true,
    sortActive: true,
    ascending: true,
    cellStyle: 'firstColumn',
   sortCallback: sortByName
  },
  {
    columnId: 'status',
    header: 'Status',
    sortable: true,
    showStatus: true,
    cellStyle: 'normalImportance',
    sortCallback: sortByUploadTime,
  }
]
```

#### Option B 
Adding a sorting type kind of the one that we talk. ISortConfig would be optional but if you use it columnId is required.

```js

type ISortConfig = {
  columnId: string,
  active?: boolean,
   isAscending?: boolean
}
```

The User would still send the sortCallback in the main props and the column config would look like this.

```js
const columnConfigSample : ITableColumnConfig[] = [
  {
    header: 'Device Name',
    cellStyle: 'firstColumn',
    sortConfig: {columnId:"deviceName", active: true, isAscending: true}
  },
  {
    header: 'Status',
    showStatus: true,
    cellStyle: 'normalImportance',
  }
]
```
Sorry for the long message in advance.
Since this will be used in many places in the future and you are familiar with the use cases I think it was worth it to discuss it more.

Reply:
I prefer you always callback with a column ID. In the case where a  is provided use that, in the case where no column id is provided by user fallback for an index based column ID, something like `column_${index}` for the default columnID might help with clarity. 


IMO you should/could set that default `columnID` once at the initial  layer and use it everywhere (not just here)